mlir/lib/Conversion/PDLToPDLInterp/PredicateTree.cpp

```diff
@@ -768,17 +768,19 @@ struct OrderedPredicate {
   /// model.
   bool operator<(const OrderedPredicate &rhs) const {
     // Sort by:
+    // * not being a constraint
```

Comment:
Could we add some reason here why we want to have constraints last? Just to remember it for the future